File: AngularLovelyjubbly.Data.Sql.Models/PlayResult.cs
using System;

namespace AngularLovelyjubbly.Data.Sql.Models
{
    public partial class PlayResult
    {
        public PlayResult()
        {
        }

        public int PlayResultId { get; set; }
        //foreign key
        public int FormationId { get; set; }
        //navigation property
        public virtual Formation Formation { get; set; }
        //foreign key
        public int OffensivePlayId { get; set; }
        //navigation property
        public virtual OffensivePlay OffensivePlay { get; set; }
        //foreign key
        public int DefensivePlayId { get; set; }
        //navigation property
        public virtual DefensivePlay DefensivePlay { get; set; }
        public int Yards { get; set; } //(inc sacks and pens)
        public Boolean IsOffensivePenalty { get; set; }
        public Boolean IsDefensivePenalty { get; set; }
        public Boolean IsSack { get; set; }
        public Boolean IsFumble { get; set; }
        public Boolean IsInterception { get; set; }
        public int ReturnYards { get; set; } //(fumble or int yards returned - yards gained)
    }
}

File: AngularLovelyjubbly/ClientApp/src/app/shared/models/playresult.ts
import { IFormation } from './formation';
import { IOffensivePlay } from './offensiveplay';
import { IDefensivePlay } from './defensiveplay';

export interface IPlayResult {
    playResultId: number;
    formation: IFormation;
    offensivePlay: IOffensivePlay;
    defensivePlay: IDefensivePlay;
    yards: number;
    isOffensivePenalty: boolean;
    isDefensivePenalty: boolean;
    isSack: boolean;
    isFumble: boolean;
    isInterception: boolean;
    returnYards: number;
}

export class PlayResult implements IPlayResult {

    constructor(public playResultId: number = 0,
        public formation: IFormation = null,
        public offensivePlay: IOffensivePlay = null,
        public defensivePlay: IDefensivePlay = null,
        public yards: number = 0,
        public isOffensivePenalty: boolean = false,
        public isDefensivePenalty: boolean = false,
        public isSack: boolean = false,
        public isFumble: boolean = false,
        public isInterception: boolean = false,
        public returnYards: number = 0) {
    }
}
